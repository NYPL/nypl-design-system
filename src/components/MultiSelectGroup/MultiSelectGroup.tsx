import { chakra, Stack } from "@chakra-ui/react";
import React, { forwardRef } from "react";

import Fieldset from "../Fieldset/Fieldset";
import { LayoutTypes } from "../../helpers/types";
import MultiSelect, { MultiSelectWidths } from "../MultiSelect/MultiSelect";
import useNYPLBreakpoints from "../../hooks/useNYPLBreakpoints";

export interface MultiSelectGroupProps {
  children: React.ReactNode;
  /** Additional className to use. */
  className?: string;
  id: string;
  labelText: string;
  /** Renders the layout of `MultiSelect` components in a row or column. */
  layout?: LayoutTypes;
  /** Width will be passed on each `MultiSelect` component. */
  multiSelectWidth?: MultiSelectWidths;
  /** Is set to `true` by default and determines if the `labelText` is visible on the site. */
  showLabel?: boolean;
}

/**
 * `MultiSelectGroup` is a wrapper component specific for `MultiSelect` components. The wrapped `MutliSelect` components can be displayed in a
 * column or in a row. The `MultiSelectGroup` component renders all the necessary
 * wrapping and associated text elements, but the child elements
 * _need_ to be `MultiSelect` components from the NYPL Design System.
 */
export const MultiSelectGroup = chakra(
  forwardRef<HTMLDivElement, React.PropsWithChildren<MultiSelectGroupProps>>(
    (props, ref?) => {
      const {
        children,
        className = "",
        id,
        labelText,
        layout = "row",
        multiSelectWidth = "default",
        showLabel = true,
        ...rest
      } = props;
      const newChildren: JSX.Element[] = [];

      const { isLargerThanMobile } = useNYPLBreakpoints();
      const finalLayout = isLargerThanMobile ? layout : "column";
      const isBlockElement = layout === "column" ? true : false;

      // Go through the MultiSelect children and update props as needed.
      React.Children.map(
        children as JSX.Element,
        (child: React.ReactElement) => {
          if (child.type !== MultiSelect) {
            console.warn(
              "NYPL Reservoir MultiSelectGroup: Only MultiSelect components can be children of MultiSelectGroup."
            );
            return;
          }
          if (child !== undefined && child !== null) {
            newChildren.push(
              React.cloneElement(child, {
                isBlockElement,
                width: multiSelectWidth,
              })
            );
          }
        }
      );

      return (
        <Fieldset
          id={`${id}-multiselect-group`}
          legendText={labelText}
          isLegendHidden={!showLabel}
          width={multiSelectWidth}
          {...rest}
        >
          <Stack
            id={id}
            ref={ref}
            data-testid="multi-select-group"
            aria-label={!showLabel ? labelText : undefined}
            className={className}
            direction={finalLayout}
            spacing="xs"
          >
            {newChildren}
          </Stack>
        </Fieldset>
      );
    }
  )
);

export default MultiSelectGroup;
