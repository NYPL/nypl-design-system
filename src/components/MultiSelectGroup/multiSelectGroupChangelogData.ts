/** This data is used to populate the ComponentChangelogTable component.
 *
 * date: string (when adding new entry during development, set value as "Prerelease")
 * version: string (when adding new entry during development, set value as "Prerelease")
 * type: "Bug Fix" | "New Feature" | "Update";
 * affects: array["Accessibility" | "Documentation" | "Functionality" | "Styles"];
 * notes: array (will render as a bulleted list, add one array element for each list element)
 */
import { ChangelogData } from "../../utils/ComponentChangelogTable";

export const changelogData: ChangelogData[] = [
  {
    date: "2024-07-03",
    version: "3.1.7",
    type: "Update",
    affects: ["Documentation"],
    notes: [
      "Added an example of a `MultiSelectGroup` with the `MultiSelect`'s `closeOnBlur` set to true",
    ],
  },
  {
    date: "2024-04-11",
    version: "3.1.0",
    type: "Update",
    affects: ["Accessibility", "Documentation", "Functionality", "Styles"],
    notes: [
      "V2 refactor of the `MultiSelectGroup` component including updating to Chakra 2.8.",
    ],
  },
];
