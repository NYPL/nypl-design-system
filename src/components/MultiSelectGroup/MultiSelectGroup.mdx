import { ArgTypes, Canvas, Description, Meta } from "@storybook/blocks";

import ComponentChangelogTable from "../../utils/ComponentChangelogTable";
import * as MultiSelectGroupStories from "./MultiSelectGroup.stories";
import Link from "../Link/Link";
import { changelogData } from "./multiSelectGroupChangelogData";

<Meta of={MultiSelectGroupStories} />

# MultiSelectGroup

| Component Version | DS Version   |
| ----------------- | ------------ |
| Added             | `1.4.0`      |
| Latest            | `Prerelease` |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#layout-patterns" target="_self">Layout Patterns</Link>}
- {<Link href="#changelog" target="_self">Changelog</Link>}

## Overview

<Description of={MultiSelectGroupStories} />

## Component Props

<Canvas of={MultiSelectGroupStories.WithControls} />

<ArgTypes of={MultiSelectGroupStories.WithControls} />

## Accessibility

The `MultiSelectGroup` renders a group of `MultiSelect` components that are wrapped
in a `<fieldset>` element. The `<fieldset>` element renders a `<legend>` element
that can be visually hidden through the `showLabel` prop. If the `<legend>` is
hidden an `aria-label` attribute with `labelText` value will be rendered instead.

For accessibility information on the `MultiSelect` component, see the
[MultiSelect Accessibility](../?path=/docs/components-form-elements-multiselect--docs#accessibility) documentation.

## Layout Patterns

The width of the MultiSelect components can be passed through the
`multiSelectWidth` prop, with values of `"fitContent"` or `"full"`.
On screen width under 600px width the `layout` is automatically set to "column"
and the `multiSelectWidth` to "full". Note: when `layout = "column"` is passed,
the MultiSelect components will have `isBlockElement` set to `true` by default.

<Canvas of={MultiSelectGroupStories.LayoutPatterns} />

## useMultiSelect hook

`useMultiSelect` is an optional hook for
cases where managing the state of the component in the consuming app is less
of a concern and general ease of use is prefered.

The hook returns an object containing all the props and state needed to handle
the selectedItems. That includes the functions `onChange`, `onClearAll`,
`onMixedStateChange` for handling any changes to the selection of items and the
current state of the selection: `selectedItems`. Additionally it returns
`setSelectedItems` for setting an initial state of `selectedItems` or manipulate
the `selectedItems` if needed.

Due to it's data structure and functions setup, it is capable to handle a group
of `MultiSelect` components. Find the full documentation under

[useMultiSelect](../?path=/docs/hooks-usemultiselect--docs).

## Changelog

<ComponentChangelogTable changelogData={changelogData} />
