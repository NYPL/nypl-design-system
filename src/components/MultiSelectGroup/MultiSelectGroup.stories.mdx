import {
  ArgsTable,
  Canvas,
  Description,
  Meta,
  Story,
} from "@storybook/addon-docs";
import { withDesign } from "storybook-addon-designs";

import DSProvider from "../../theme/provider";
import { getCategory } from "../../utils/componentCategories";
import MultiSelectGroup from "./MultiSelectGroup";
import * as stories from "./MultiSelectGroup.stories";

<Meta
  title={getCategory("MultiSelectGroup")}
  component={MultiSelectGroup}
  decorators={[withDesign]}
  parameters={{
    design: {
      type: "figma",
      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=43593%3A24611",
    },
    jest: ["MultiSelectGroup.test.tsx"],
  }}
  argTypes={{
    className: { control: false },
    id: { control: false },
    isBlockElement: {
      control: "radio",
      options: [true, false, undefined],
      table: { defaultValue: { summary: undefined } },
    },
    layout: {
      table: { defaultValue: { summary: "row" } },
    },
    showLabel: {
      table: { defaultValue: { summary: true } },
    },
    multiSelectWidth: {
      table: { defaultValue: { summary: "default" } },
    },
  }}
/>

# MultiSelectGroup

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `1.1.0`    |
| Latest            | `1.1.0`    |

## Table of Contents

- [Overview](#overview)
- [Component Props](#component-props)
- [Accessibility](#accessibility)
- [Layout Patterns](#layout-patterns)

## Overview

<Description of={MultiSelectGroup} />

## Component Props

<Canvas withToolbar>
  <Story
    name="MultiSelectGroup"
    args={{
      id: "multiselect-group",
      labelText: "Label Text",
    }}
  >
    {(args) => <stories.MultiSelectGroupStory {...args} />}
  </Story>
</Canvas>

<ArgsTable story="MultiSelectGroup" />

## Accessibility

The `MultiSelectGroup` renders a group of `MultiSelect` components that are wrapped in
a `<fieldset>` element. The `<fieldset>` element renders a `<legend>` element that
can be visually hidden through the `showLabel` prop. If the `<legend>` is hidden an `aria-label`
attribute with `labelText` value will be rendered instead.
For accessibility information on the `MultiSelect` component, see the [MultiSelect Accessibility](https://nypl-design-system-vvscfj2pn-nypl.vercel.app/?path=/docs/components-form-elements-multiselect--multi-select-list-box) documentation.

## Layout Patterns

The width of the MultiSelect components can be passed through the `multiSelectWidth`
prop, with values of `"default"`, `"fitContent"`, and `"full"`.
On screen width under 600px width the `layout` is automatically set to "column" and the `multiSelectWidth` to "full".
Note: when `layout = "column"` is passed, the MultiSelect components will have `isBlockElement` set to `true` by default.

<Canvas>
  <DSProvider>
    <stories.MultiSelectGroupLayoutStory />
  </DSProvider>
</Canvas>
