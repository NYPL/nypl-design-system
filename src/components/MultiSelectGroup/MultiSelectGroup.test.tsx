import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import * as React from "react";
import renderer from "react-test-renderer";

import MultiSelectGroup from "./MultiSelectGroup";
import MultiSelect from "../MultiSelect/MultiSelect";

const multiSelectItems = [
  {
    id: "colors",
    name: "Colors",
    items: [
      { id: "red", name: "Red" },
      { id: "blue", name: "Blue" },
      { id: "yellow", name: "Yellow" },
    ],
  },
  {
    id: "pets",
    name: "Pets",
    items: [
      { id: "cat", name: "Cat" },
      {
        id: "dog",
        name: "Dog",
        children: [
          { id: "corgy", name: "Corgy" },
          { id: "german-sheperd", name: "German Sheperd" },
          { id: "afghan-hound", name: "Afghan Hound" },
        ],
      },
      { id: "rat", name: "Rat" },
    ],
  },
  {
    id: "tools",
    name: "Tools",
    items: [
      { id: "hammer", name: "Hammer" },
      {
        id: "skrewdriver",
        name: "Skrewdriver",
        children: [
          { id: "slottet", name: "Slotted Skrewdriver" },
          { id: "phillips", name: "Phillips Skrewdriver" },
          { id: "allen", name: "Allen Skrewdriver" },
        ],
      },
      { id: "whisk", name: "Whisk" },
    ],
  },
];

const defaultItemsVisible = 5;

describe.skip("MulitSelectGroup Accessibility", () => {
  it("passes axe accessibility with string labels ", async () => {
    const handleChangeMock = jest.fn();
    const { container } = render(
      <MultiSelectGroup
        id="MultiSelectGroup"
        labelText="MultiSelectGroup example"
        showLabel={true}
        multiSelectWidth="full"
      >
        {multiSelectItems.map((multiSelectItem) => (
          <MultiSelect
            key={multiSelectItem.id}
            id={multiSelectItem.id}
            items={multiSelectItem.items}
            selectedItems={{}}
            isDefaultOpen={false}
            isSearchable={false}
            isBlockElement={false}
            buttonText="MultiSelect"
            defaultItemsVisible={defaultItemsVisible}
            onChange={handleChangeMock}
            onClear={() => "clear"}
          />
        ))}
      </MultiSelectGroup>
    );
    expect(await axe(container)).toHaveNoViolations();
  });
  it("passes axe accessibility with the legend hidden", async () => {
    const handleChangeMock = jest.fn();
    const { container } = render(
      <MultiSelectGroup
        id="MultiSelectGroup"
        labelText="MultiSelectGroup example"
        showLabel={false}
        multiSelectWidth="full"
      >
        {multiSelectItems.map((multiSelectItem) => (
          <MultiSelect
            key={multiSelectItem.id}
            id={multiSelectItem.id}
            items={multiSelectItem.items}
            selectedItems={{}}
            isDefaultOpen={false}
            isSearchable={false}
            isBlockElement={false}
            buttonText="MultiSelect"
            defaultItemsVisible={defaultItemsVisible}
            onChange={handleChangeMock}
            onClear={() => "clear"}
          />
        ))}
      </MultiSelectGroup>
    );
    expect(await axe(container)).toHaveNoViolations();
  });
  it("<legend> element is available in the DOM when 'showLabel' prop is set to true or false", () => {
    const handleChangeMock = jest.fn();
    const { container, rerender } = render(
      <MultiSelectGroup
        id="MultiSelectGroup"
        labelText="MultiSelectGroup example"
        showLabel={true}
        multiSelectWidth="full"
      >
        {multiSelectItems.map((multiSelectItem) => (
          <MultiSelect
            key={multiSelectItem.id}
            id={multiSelectItem.id}
            items={multiSelectItem.items}
            selectedItems={{}}
            isDefaultOpen={false}
            isSearchable={false}
            isBlockElement={false}
            buttonText="MultiSelect"
            defaultItemsVisible={defaultItemsVisible}
            onChange={handleChangeMock}
            onClear={() => "clear"}
          />
        ))}
      </MultiSelectGroup>
    );
    expect(container.querySelector("legend")).toBeVisible();
    expect(container.querySelector("legend")).toHaveTextContent(
      "MultiSelectGroup example"
    );

    rerender(
      <MultiSelectGroup
        id="MultiSelectGroup"
        labelText="MultiSelectGroup example"
        showLabel={false}
        multiSelectWidth="full"
      >
        {multiSelectItems.map((multiSelectItem) => (
          <MultiSelect
            key={multiSelectItem.id}
            id={multiSelectItem.id}
            items={multiSelectItem.items}
            isDefaultOpen={false}
            isSearchable={false}
            isBlockElement={false}
            selectedItems={{}}
            buttonText="MultiSelect"
            defaultItemsVisible={defaultItemsVisible}
            onChange={handleChangeMock}
            onClear={() => "clear"}
          />
        ))}
      </MultiSelectGroup>
    );
    expect(container.querySelector("legend")).toBeVisible();
    expect(container.querySelector("legend")).toHaveTextContent(
      "MultiSelectGroup example"
    );
  });
  xit("should throw warning when a non-MultiSelect component is used as a child", () => {
    const warn = jest.spyOn(console, "warn");
    render(
      <MultiSelectGroup
        labelText="Not a MultiSelect child!"
        showLabel={true}
        id="wrong-child"
        multiSelectWidth="full"
      >
        <p>I pretend to be a MultiSelect!</p>
      </MultiSelectGroup>
    );
    expect(warn).toHaveBeenCalledWith(
      "NYPL Reservoir MultiSelectGroup: Only MultiSelect components can be children of MultiSelectGroup."
    );
  });
  it("renders the UI snapshots correctly", () => {
    const handleChangeMock = jest.fn();

    const multiSelectGroup = renderer
      .create(
        <MultiSelectGroup
          id="MultiSelectGroup"
          multiSelectWidth="full"
          labelText="MultiSelectGroup example"
          showLabel={true}
        >
          {multiSelectItems.map((multiSelectItem) => (
            <MultiSelect
              key={multiSelectItem.id}
              id={multiSelectItem.id}
              items={multiSelectItem.items}
              selectedItems={{}}
              isDefaultOpen={false}
              isSearchable={false}
              isBlockElement={false}
              buttonText="MultiSelect"
              defaultItemsVisible={defaultItemsVisible}
              onChange={handleChangeMock}
              onClear={() => "clear"}
            />
          ))}
        </MultiSelectGroup>
      )
      .toJSON();

    const labelHidden = renderer
      .create(
        <MultiSelectGroup
          id="MultiSelectGroup"
          multiSelectWidth="full"
          labelText="MultiSelectGroup example"
          showLabel={false}
        >
          {multiSelectItems.map((multiSelectItem) => (
            <MultiSelect
              key={multiSelectItem.id}
              id={multiSelectItem.id}
              items={multiSelectItem.items}
              selectedItems={{}}
              isDefaultOpen={false}
              isSearchable={false}
              isBlockElement={false}
              buttonText="MultiSelect"
              defaultItemsVisible={defaultItemsVisible}
              onChange={handleChangeMock}
              onClear={() => "clear"}
            />
          ))}
        </MultiSelectGroup>
      )
      .toJSON();

    expect(multiSelectGroup).toMatchSnapshot();
    expect(labelHidden).toMatchSnapshot();
  });
});
