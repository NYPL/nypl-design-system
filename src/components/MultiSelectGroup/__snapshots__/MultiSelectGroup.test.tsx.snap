// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MulitSelectGroup Accessibility renders the UI snapshots correctly 1`] = `
<fieldset
  className="css-1xdhyk6"
  id="MultiSelectGroup-multiselect-group"
>
  <legend>
    MultiSelectGroup example
  </legend>
  <div
    className="chakra-stack css-ofymhf"
    data-testid="multi-select-group"
    id="MultiSelectGroup"
  >
    <div
      className="css-hhryjt"
      id="colors"
      type="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="chakra-button css-8bx9xp"
          id="ms-colors-menu-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="css-0"
            title="Colors"
          >
            Colors
          </span>
          <svg
            aria-hidden={true}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            id="ms-colors-icon"
            role="img"
            title="plus icon"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </button>
        <div
          aria-labelledby="ms-colors-menu-button"
          className="css-0"
          role="dialog"
        >
          <ul
            className="css-10ns9zz"
            role="list"
          />
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
    <div
      className="css-hhryjt"
      id="pets"
      type="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="chakra-button css-8bx9xp"
          id="ms-pets-menu-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="css-0"
            title="Pets"
          >
            Pets
          </span>
          <svg
            aria-hidden={true}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            id="ms-pets-icon"
            role="img"
            title="plus icon"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </button>
        <div
          aria-labelledby="ms-pets-menu-button"
          className="css-0"
          role="dialog"
        >
          <ul
            className="css-10ns9zz"
            role="list"
          />
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
    <div
      className="css-hhryjt"
      id="tools"
      type="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="chakra-button css-8bx9xp"
          id="ms-tools-menu-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="css-0"
            title="Tools"
          >
            Tools
          </span>
          <svg
            aria-hidden={true}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            id="ms-tools-icon"
            role="img"
            title="plus icon"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </button>
        <div
          aria-labelledby="ms-tools-menu-button"
          className="css-0"
          role="dialog"
        >
          <ul
            className="css-10ns9zz"
            role="list"
          />
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>
</fieldset>
`;

exports[`MulitSelectGroup Accessibility renders the UI snapshots correctly 2`] = `
<fieldset
  className="css-1xdhyk6"
  id="MultiSelectGroup-multiselect-group"
>
  <legend>
    MultiSelectGroup example
  </legend>
  <div
    className="chakra-stack css-a2evad"
    data-testid="multi-select-group"
    id="MultiSelectGroup"
  >
    <div
      className="css-hhryjt"
      id="colors"
      type="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="chakra-button css-8bx9xp"
          id="ms-colors-menu-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="css-0"
            title="Colors"
          >
            Colors
          </span>
          <svg
            aria-hidden={true}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            id="ms-colors-icon"
            role="img"
            title="plus icon"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </button>
        <div
          aria-labelledby="ms-colors-menu-button"
          className="css-0"
          role="dialog"
        >
          <ul
            className="css-10ns9zz"
            role="list"
          />
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
    <div
      className="css-hhryjt"
      id="pets"
      type="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="chakra-button css-8bx9xp"
          id="ms-pets-menu-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="css-0"
            title="Pets"
          >
            Pets
          </span>
          <svg
            aria-hidden={true}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            id="ms-pets-icon"
            role="img"
            title="plus icon"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </button>
        <div
          aria-labelledby="ms-pets-menu-button"
          className="css-0"
          role="dialog"
        >
          <ul
            className="css-10ns9zz"
            role="list"
          />
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
    <div
      className="css-hhryjt"
      id="tools"
      type="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <button
          className="chakra-button css-8bx9xp"
          id="ms-tools-menu-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="css-0"
            title="Tools"
          >
            Tools
          </span>
          <svg
            aria-hidden={true}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            id="ms-tools-icon"
            role="img"
            title="plus icon"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </button>
        <div
          aria-labelledby="ms-tools-menu-button"
          className="css-0"
          role="dialog"
        >
          <ul
            className="css-10ns9zz"
            role="list"
          />
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>
</fieldset>
`;
